qute.jdt/com.redhat.qute.jdt/src/main/java/com/redhat/qute/jdt/internal/template/datamodel/TemplateGlobalAnnotationSupport.java

```diff
@@ -0,0 +1,154 @@
+package com.redhat.qute.jdt.internal.template.datamodel;
+
+import static com.redhat.qute.jdt.internal.QuteJavaConstants.TEMPLATE_GLOBAL_ANNOTATION;
+import static com.redhat.qute.jdt.internal.QuteJavaConstants.TEMPLATE_GLOBAL_ANNOTATION_NAME;
+
+import java.lang.reflect.Modifier;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.IAnnotation;
+import org.eclipse.jdt.core.IField;
+import org.eclipse.jdt.core.IJavaElement;
+import org.eclipse.jdt.core.IMember;
+import org.eclipse.jdt.core.IMethod;
+import org.eclipse.jdt.core.IType;
+import org.eclipse.jdt.core.JavaModelException;
+
+import com.redhat.qute.commons.datamodel.resolvers.ValueResolverInfo;
+import com.redhat.qute.jdt.QuteSupportForTemplate;
+import com.redhat.qute.jdt.internal.resolver.ITypeResolver;
+import com.redhat.qute.jdt.template.datamodel.AbstractAnnotationTypeReferenceDataModelProvider;
+import com.redhat.qute.jdt.template.datamodel.SearchContext;
+import com.redhat.qute.jdt.utils.AnnotationUtils;
+
+/**
+ * @TemplateGlobal annotation support.
+ *
+ *                 <code>
+ * &#64;TemplateGlobal
+ * public class Globals {
+ *
+ * 		static int age = 40;
+ *
+ * 		static Color[] myColors() {
+ * 			return new Color[] { Color.RED, Color.BLUE };
+ * 		}
+ *
+ * 		&#64;TemplateGlobal(name = "currentUser")
+ * 		static String user() {
+ * 			return "Mia";
+ * 		}
+ * }
+ * </code>
+ *
+ *
+ * @see https://quarkus.io/guides/qute-reference#global_variables
+ *
+ */
+public class TemplateGlobalAnnotationSupport extends AbstractAnnotationTypeReferenceDataModelProvider {
+
+	private static final Logger LOGGER = Logger.getLogger(TemplateGlobalAnnotationSupport.class.getName());
+
+	private static final String[] ANNOTATION_NAMES = {
+		TEMPLATE_GLOBAL_ANNOTATION
+	};
+
+	@Override
+	protected String[] getAnnotationNames() {
+		return ANNOTATION_NAMES;
+	}
+
+	@Override
+	protected void processAnnotation(IJavaElement javaElement, IAnnotation annotation, String annotationName,
+		SearchContext context, IProgressMonitor monitor) throws JavaModelException {
+		if (javaElement.getElementType() != IJavaElement.TYPE) {
+			return;
+		}
+		IType type = (IType) javaElement;
+		String name = null;
+		boolean hasTemplateGlobal = false;
+		for (IAnnotation typeAnnotation : type.getAnnotations()) {
+			if (AnnotationUtils.isMatchAnnotation(typeAnnotation, TEMPLATE_GLOBAL_ANNOTATION)) {
+				hasTemplateGlobal = true;
+				name = AnnotationUtils.getAnnotationMemberValue(typeAnnotation, TEMPLATE_GLOBAL_ANNOTATION_NAME);
+				if (StringUtils.isNotEmpty(name)) {
+					break;
+				}
+			}
+		}
+		if (!hasTemplateGlobal) {
+			return;
+		}
+
+		ITypeResolver typeResolver = QuteSupportForTemplate.createTypeResolver(type);
+
+		// Loop for static fields
+		IField[] fields = type.getFields();

```

Comment:
I suggest that you read Javadoc at https://github.com/quarkusio/quarkus/blob/5e6fe630375c96b18228f54c6537f445f3ee1743/independent-projects/qute/core/src/main/java/io/quarkus/qute/TemplateGlobal.java#L10 which explains rules that field and method must match (ex : static, etc) and implement those rules.

Reply:
So both a field and method must be non private and static, and a method must have no parameters and return void, correct?